cmlreaders/readers/readers.py

```diff
@@ -149,19 +152,54 @@ class MontageReader(BaseCMLReader):
     @staticmethod
     def _flatten_row(data: dict, labels: List[str], i: int) -> pd.DataFrame:
         entry = data[labels[i]].copy()
-        atlases = entry.pop('atlases')
-        atlas_row = json_normalize(atlases)
-        atlas_row.index = [i]
-        row = pd.concat([pd.DataFrame(entry, index=[i]), atlas_row], axis=1)
-        return row
+        try:
+            atlases = entry.pop('atlases')
+            atlas_row = json_normalize(atlases)
+            atlas_row.index = [i]
+            row = pd.concat([pd.DataFrame(entry, index=[i]), atlas_row], axis=1)
+            return row
+        except KeyError:
+            # Ramulator pairs.json files are more minimal and do not include
+            # the atlas locations
+            return pd.DataFrame(entry, index=[i])
 
     def as_dataframe(self):
         from concurrent.futures import ProcessPoolExecutor as Pool
         from functools import partial
         from multiprocessing import cpu_count
 
-        with open(self._file_path) as f:
-            data = json.load(f)[self.subject][self.data_type]
+        if self.data_type == "pairs":
+            try:
+                rerefable = is_rerefable(self.subject, self.experiment,
+                                         self.session, self.localization,
+                                         self.montage, self.rootdir)
+            except FileNotFoundError:
+                # We get this error when testing with file_path specified, so
+                # don't worry about it here.
+                import warnings
+                warnings.warn(
+                    "Couldn't determine if EEG data can be rereferenced. "
+                    "This can happen if you didn't specify an experiment and/or"
+                    " session, or if you're passing file_path explicitly. "
+                    "Proceeding with the assumption that rereferencing is "
+                    "possible.", UserWarning
+                )
+                rerefable = True
+        else:
+            rerefable = True  # for contacts, we don't care about this
+
+        if rerefable:
+            with open(self._file_path) as f:
+                data = json.load(f)[self.subject][self.data_type]
+        else:
+            # look for the pairs.json file actually used by Ramulator
```

Comment:
have always been producing these pairs.json files? I'm wondering if they won't be available for some of the older subjects, but maybe that's not a problem since it is really the system 3 hardware bipolar subjects that we need to be concerned about here, and I believe all of those will have the minimal pairs.json file

Reply:
Ramulator has always used `pairs.json` as far as I am aware. The `is_rerefable` function means that it won't go looking for that unless it has to, at any rate.